### PR TITLE
Relative translation scopes

### PR DESCRIPTION
**Pull request by elia**
### Summary

Add support for relative scopes, i.e. scopes starting with a `.` that will be prefixed to the translation key and then chained to the component base scope.

Fixes https://github.com/github/view_component/issues/1334

### CODE REVIEW

File: docs/CHANGELOG.md [hunk 1]
```diff
@@ -9,6 +9,10 @@ title: Changelog
 
 ## main
 
+* Add support for relative I18n scopes to translations
```
Comment (camertron) [inline docs/CHANGELOG.md:12]:
```suggestion
* Add support for relative I18n scopes to translations.
```

**Review by camertron — COMMENTED** [review]
Looking good! 🎉  Could you add a test for the array case? Thanks!

**Review by joelhawksley — APPROVED** [review]

File: test/view_component/translatable_test.rb [hunk 2]
```diff
@@ -122,6 +122,15 @@ def test_translate_returns_html_safe_lists
     end
   end
 
+  def test_translate_scopes
+    assert_equal "This is coming from the sidecar", translate("sidecar", scope: '.from')
+    assert_equal "This is coming from Rails", translate("rails", scope: 'from')
+    assert_equal "This is coming from the sidecar", translate("sidecar", scope: ['.from'])
+    assert_equal "This is coming from Rails", translate("rails", scope: ['from'])
+    assert_equal "This is coming from the sidecar", translate(:sidecar, scope: [:'.from'])
+    assert_equal "This is coming from Rails", translate(:rails, scope: [:from])
```
Comment (joelhawksley) [inline test/view_component/translatable_test.rb:131]:
```suggestion
    assert_equal "This is coming from the sidecar", translate("sidecar", scope: ".from")
    assert_equal "This is coming from Rails", translate("rails", scope: "from")
    assert_equal "This is coming from the sidecar", translate("sidecar", scope: [".from"])
    assert_equal "This is coming from Rails", translate("rails", scope: ["from"])
    assert_equal "This is coming from the sidecar", translate(:sidecar, scope: [:".from"])
    assert_equal "This is coming from Rails", translate(:rails, scope: [:from])
```

File: lib/view_component/translatable.rb [hunk 3]
```diff
@@ -68,7 +67,10 @@ def translate(key = nil, **options)
       return key.map { |k| translate(k, **options) } if key.is_a?(Array)
 
       locale = options.delete(:locale) || ::I18n.locale
+      scope = options.delete(:scope)
+      scope = scope.join('.') if scope.is_a? Array
```
Comment (joelhawksley) [inline lib/view_component/translatable.rb:71]:
```suggestion
      scope = scope.join(".") if scope.is_a? Array
```

**Pull request merged**